tasks/main.yml

```diff
@@ -17,26 +12,237 @@
+      - "mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula"
+      - "mssql_accept_microsoft_cli_utilities_for_sql_server_eula"
+      - "mssql_accept_microsoft_sql_server_2019_standard_eula"
+  register: __mssql_eula_evaluation
 
-# Examples of some tasks:
-- name: Ensure required packages are installed
+- name: Ensure that the tuned-profiles-mssql package is installed
   package:
-    name: "{{ __template_packages }}"
+    name: tuned-profiles-mssql
     state: present
 
-- name: Ensure required services are enabled and started
+- name: Ensure that the tuned service is started and enabled
   service:
-    name: "{{ item }}"
+    name: tuned
     state: started
-    enabled: yes
-  loop: "{{ __template_services }}"
-
-- name: Generate /etc/{{ __template_foo_config }}
-  template:
-    src: "{{ __template_foo_config }}.j2"
-    dest: /etc/{{ __template_foo_config }}
-    backup: yes
-    mode: 0400
-  notify: template handler to restart services
+    enabled: true
+
+- name: Get the active Tuned profile
+  command: tuned-adm active
+  changed_when: false
+  register: __mssql_tuned_active_profile
+
+- name: Set the mssql Tuned profile
+  command: tuned-adm profile mssql
+  when: '"mssql" not in __mssql_tuned_active_profile.stdout'
+
+- name: Configure the Microsoft SQL Server 2019 repository
+  yum_repository:
+    name: packages-microsoft-com-mssql-server-2019-gdr
+    description: Microsoft SQL Server 2019
+    baseurl: "{{ __mssql_server_repository }}"
+    gpgkey: https://packages.microsoft.com/keys/microsoft.asc
+    gpgcheck: true
+
+- name: Ensure that the Microsoft SQL Server package is installed
+  package:
+    name: "{{ __mssql_server_packages }}"
+    state: present
+
+- name: Check if the errorlog file exists and its location
+  shell: |
+    set -euo pipefail
+    errorlog="$(/opt/mssql/bin/mssql-conf get filelocation errorlogfile 2>&1 \
+    | sed 's/errorlogfile : //')" || :
+    if [ -f "${errorlog}" ]; then
+      echo "${errorlog}"
+    elif [ -f /var/opt/mssql/log/errorlog ]; then
+      echo /var/opt/mssql/log/errorlog
+    else
+      echo ""
+    fi
+  changed_when: false
+  register: __mssql_errorlog
+
+- name: Gather system services facts
+  service_facts:
+
+- name: Determine if MSSQL was previously set up
+  set_fact:
+    __mssql_is_setup: >-
+      {{ ('running' in
+      ansible_facts['services']['mssql-server.service']['state']) or
+      ('enabled' in
+      ansible_facts['services']['mssql-server.service']['status']) or
+      (__mssql_errorlog.stdout | length > 0) }}
+
+- name: Verify that the variables required for setting up MSSQL are defined
+  assert:
+    that:
+      - mssql_password is defined
+      - mssql_edition is defined
+    fail_msg:
+      - "You must define the following variables to set up MSSQL:"
+      - "mssql_password"
+      - "mssql_edition"
+  when: not __mssql_is_setup
+  register: __mssql_variables_evaluation
```

Comment:
You don't need to register this.  Instead, the fail will "throw" an exception that you can catch in a `rescue` block and examine the message with `ansible_failed_result.results.0.msg` - see https://github.com/linux-system-roles/logging/blob/master/tests/tests_basics_forwards.yml#L542-L574 for an example of a role that fails with a specific error, and a tests that checks if the role failed with that specific error message